src/data/finset/pointwise.lean

```diff
@@ -78,6 +78,14 @@ localized "attribute [instance] finset.has_one finset.has_zero" in pointwise
 @[simp, to_additive]
 lemma image_one [decidable_eq β] {f : α → β} : image f 1 = {f 1} := image_singleton f 1
 
+/-- The singleton operation as a `one_hom`. -/
+@[to_additive "The singleton operation as a `zero_hom`."]
+def singleton_one_hom : one_hom α (finset α) := ⟨singleton, singleton_one⟩
+
+@[simp, to_additive] lemma coe_singleton_one_hom : (singleton_one_hom : α → finset α) = singleton :=
+rfl
+@[simp, to_additive] lemma singleton_one_hom_apply (a : α) : singleton_one_hom a = {a} := rfl
```

Comment:
Does `simps` generate this one?

Reply:
Yes, but generating it with `simps` means that `coe_singleton_one_hom` is declared after and has priority, which means it's not simp normal form.

Reply:
I'd argue already that the LHS of `singleton_one_hom_apply` is not simp-normal form, but perhaps this case is deliberately allowed by `simp_nf` and therefore it doesn't matter. I hadn't realized it cared about order in this way.

Reply:
Yeah, one would think that the `coe_` lemma is enough, but it is not applied so sometimes it doesn't fire (or at least that's what I perceived. This is why all homs have the `coe_hom`, `hom_apply` pair, *in that order*. We might want to review this at some point because it is an obscure technicality.